src/pretty/flow_like.rs

```diff
@@ -0,0 +1,114 @@
+use pretty::{Arena, DocAllocator};
+use typst_syntax::{SyntaxKind, SyntaxNode};
+
+use super::{util::is_comment_node, ArenaDoc, PrettyPrinter};
+
+pub struct FlowItem<'a>(Option<FlowItemRepr<'a>>);
+
+struct FlowItemRepr<'a> {
+    doc: ArenaDoc<'a>,
+    /// Whether a space can be added before the doc. Not enforcement.
+    space_before: bool,
+    ///  Whether a space can be added after the doc.
```

Comment:
```suggestion
    /// Whether a space can be added after the doc.
```

Reply:
Is it enforced or not?

Reply:
I did not make it clear enough. Actually, a space is added only when the item before and the item after both allow this.